monitoring/src/listener/records.rs

```diff
@@ -0,0 +1,149 @@
+        Self { db }
+    }
+
+    #[inline]
+    pub fn put_record_meta(&mut self, ts: f32, record_meta: &RecordMeta) -> Result<(), StorageError> {
+        self.db.put(&RocksStamp(ts), record_meta)
+            .map_err(StorageError::from)
+    }
+}
+
+impl Schema for RecordMetaStorage {
+    const COLUMN_FAMILY_NAME: &'static str = "record_meta_storage";
+    type Key = RocksStamp;
+    type Value = RecordMeta;
+}
+
+#[derive(Clone)]
+pub struct RecordStorage {
+    db: Arc<RecordStorageDatabase>,
+}
+
+impl RecordStorage {
+    pub fn new(db: Arc<RecordStorageDatabase>) -> Self {
+        Self { db }
+    }
+
+    pub fn put_record(&mut self, ts: f32, record: &Vec<u8>) -> Result<(), StorageError> {
+        self.db.put(&RocksStamp(ts), record)
+            .map_err(StorageError::from)
+    }
+}
+
+impl Schema for RecordStorage {
+    const COLUMN_FAMILY_NAME: &'static str = "record_storage";
+    type Key = RocksStamp;
+    type Value = Vec<u8>;
+}
+
+// --- Record implementation --- //
+
+/// Simple timestamp, denoting passed seconds from start of the node
+/// implementing the codec trait for simple database serialization.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct RocksStamp(f32);
+
+impl From<f32> for RocksStamp {
+    fn from(val: f32) -> Self {
+        Self(val)
+    }
+}
+
+impl Codec for RocksStamp {
+    #[inline]
+    fn decode(data: &[u8]) -> Result<Self, SchemaError> {
+        if data.len() < 4 {
+            Err(SchemaError::DecodeError)
+        } else {
+            Ok(Self((&data[0..4]).read_f32::<LittleEndian>().map_err(|_| SchemaError::DecodeError)?))
+        }
+    }
+
+    #[inline]
+    fn encode(&self) -> Result<Vec<u8>, SchemaError> {
+        let mut data = Vec::with_capacity(4);
+        data.write_f32::<LittleEndian>(self.0).map_err(|_| SchemaError::EncodeError)?;
+        Ok(data)
+    }
+}
+
+/// Part of the Record Meta, denoting content type in the Record storage
+#[derive(Debug, Copy, Clone)]
+#[repr(u8)]
+pub enum RecordType {
+    /// Connection to new peer was accepted. No record in Record Storage should correspond to
+    /// this meta message
+    PeerCreated,
+    /// Full connection to peer was established. Record should contains a string with public
+    /// key of connected peer.
+    PeerBootstrapped,
+    /// Node received a message from a peer. Record should contain raw message, which
+    /// should be deserialized manually.
+    PeerReceivedMessage,
+}
+
+impl RecordType {
+    fn from_u8(val: u8) -> Option<Self> {
+        use RecordType::*;
+        if val == PeerCreated as u8 {
+            Some(PeerCreated)
+        } else if val == PeerBootstrapped as u8 {
+            Some(PeerBootstrapped)
+        } else if val == PeerReceivedMessage as u8 {
+            Some(PeerReceivedMessage)
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+/// Record metadata, describing incoming message.
+pub struct RecordMeta {
```

Comment:
did you split the record value into 2 structures as a optimalization?

Because the
```
pub struct RecordMeta {
    /// Description of type of incoming message, and stored format.
    pub record_type: RecordType,
    /// Representation of an peer actor, to whom belongs the message
    pub peer_id: String,
}
```

and the 

```
pub enum RecordType {
    /// Connection to new peer was accepted. No record in Record Storage should correspond to
    /// this meta message
    PeerCreated,
    /// Full connection to peer was established. Record should contains a string with public
    /// key of connected peer.
    PeerBootstrapped,
    /// Node received a message from a peer. Record should contain raw message, which
    /// should be deserialized manually.
    PeerReceivedMessage,
}
```

could be replaced by something like this:
```
pub enum RecordType {
    /// Connection to new peer was accepted. No record in Record Storage should correspond to
    /// this meta message
    PeerCreated { peer_id: String },
    /// Full connection to peer was established. Record should contains a string with public
    /// key of connected peer.
    PeerBootstrapped { peer_id: String },
    /// Node received a message from a peer. Record should contain raw message, which
    /// should be deserialized manually.
    PeerReceivedMessage { peer_id: String },
}
```